### PR TITLE
Support SparkXShards for PyTorch Ray Estimator

### LINKED ISSUE #3048: PyTorch Ray Estimator fit doesn't have batch_size argument

### PR DESCRIPTION
**Pull request by hkvision**
Similar to MXNet, support SparkXShards of RDD[dict of numpy] where each records contains 'x'  for features and 'y' for labels. Features and labels can be ndarray, list/dict/tuple of ndarrays.

This PR first focuses on training fit. If the implementation is OK, then I will modify validate as well.

Also fixes #3048 Add documentation that batch size should be batch size per worker.

- [x] Add unit tests.

- [x] Update validate.

- [x] Remove print/debug messages.

### CODE REVIEW

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.210:18888/job/ZOO-PR-Validation/4504/

File: pyzoo/zoo/orca/data/utils.py [hunk 1]
```diff
@@ -143,55 +143,46 @@ def _flatten_xy(data):
...
-        return [np.concatenate((list1[index], list2[index]), axis=0)
-                for index in range(0, len(list1))]
-
-    def combine_tuple(tuple1, tuple2):
-        return tuple(np.concatenate((tuple1[index], tuple2[index]), axis=0)
-                     for index in range(0, len(tuple1)))
-
-    def check_type_and_combine(data_list):
-        if isinstance(data_list[0], dict):
-            return reduce(lambda dict1, dict2: combine_dict(dict1, dict2), data_list)
```
Comment (hkvision) [inline pyzoo/zoo/orca/data/utils.py:172]: Using reduce is really slow and even gets crash on large data partitions. Changing to list comprehension becomes much faster.

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.210:18888/job/ZOO-PR-Validation/4536/
I will merge it first. Feel free to raise comments and I will address in a following PR.
